api/server.go

```diff
@@ -0,0 +1,97 @@
+// Package server provides a server object that represents the GoCert backend
+package server
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/canonical/gocert/internal/certdb"
+	"gopkg.in/yaml.v3"
+)
+
+type ConfigYAML struct {
+	KeyPath  string
+	CertPath string
+	DBPath   string
+	Port     int
+}
+
+type Config struct {
+	Key    []byte
+	Cert   []byte
+	DBPath string
+	Port   int
+}
+
+type Environment struct {
+	DB *certdb.CertificateRequestsRepository
+}
```

Comment:
I actually had started off with db directly, but when I searched online for different ways to inject variables into handlers, people were doing the pattern that I was doing for environment variables since there's no global environment variable. Right now it's just DB, but it could be any environment variable that the handlers may need to be dynamically loaded, for example some config values from the configuration file like constraints to the certs could be placed here.